config/config.go

```diff
@@ -161,6 +166,160 @@ type AuthConfig struct {
+// Precisely one of JWT, Any and All should be set.
+type AuthenticationRequirement struct {
+	// JWT defines the JWTAuthentication
+	JWT JWTAuthentication `yaml:"jwt,omitempty" json:"jwt,omitempty"`
+
+	// Any contains a list of AuthenticationRequirements.
+	// A successful authentication requires one of them being satisfied.
+	Any []AuthenticationRequirement `yaml:"any,omitempty" json:"any,omitempty"`
+
+	// All contains a list of AuthenticationRequirements.
+	// A successful authentication requires all of them being satisfied.
+	All []AuthenticationRequirement `yaml:"all,omitempty" json:"all,omitempty"`
+}
+
+// JWTAuthentication defines the JWT authentication
+type JWTAuthentication struct {
+	// Name of this JWT Provider, unique within the Proxy.
+	Name string `yaml:"name" json:"name"`
+
+	// JWT issuer ("iss" claim)
+	Issuer string `yaml:"issuer" json:"issuer"`
+
+	// A remote JWKS source
+	RemoteJWKS RemoteJWKS `yaml:"remote_jwks" json:"remote_jwks"`
+
+	// Audiences contains a list of audiences
+	Audiences []string `yaml:"audiences,omitempty" json:"audiences,omitempty"`
+
+	// Header name that will contain decoded JWT payload in requests forwarded to
+	// target.
+	ForwardPayloadHeader string `yaml:"forward_payload_header,omitempty" json:"forward_payload_header,omitempty"`
+
+	// Locations where JWT may be found. First match wins.
+	In []HTTPParameter `yaml:"in" json:"in"`
+}
+
+// RemoteJWKS contains information for remote JWKS
+type RemoteJWKS struct {
+	// URL of the JWKS
+	URL string `yaml:"url" json:"url"`
+
+	// CacheDuration of the JWKS
+	CacheDuration time.Duration `yaml:"cache_duration,omitempty" json:"cache_duration,omitempty"`
+}
+
+// ConsumerAuthorization is the configuration of API consumer authorization
+type ConsumerAuthorization struct {
+	FailOpen bool `yaml:"fail_open,omitempty" json:"fail_open,omitempty"`
+
+	// Locations of API consumer credential (API Key). First match wins.
+	In []HTTPParameter `yaml:"in" json:"in"`
+}
+
+// HTTPMatch is an HTTP request matching rule
+type HTTPMatch struct {
+	// URL path template using to match incoming requests and optionally identify
+	// path variables.
+	PathTemplate string `yaml:"path_template" json:"path_template"`
+
+	// HTTP method (e.g. GET, POST, PUT, etc.)
+	Method string `yaml:"method,omitempty" json:"method,omitempty"`
+}
+
+// HTTPParameter defines an HTTP paramter
+// Precisely one of Query, Header or JWTClaim should be set.
+type HTTPParameter struct {
+	// Name of a query paramter
+	Query string `yaml:"query,omitempty" json:"query,omitempty"`
+
+	// Name of a header
+	Header string `yaml:"header,omitempty" json:"header,omitempty"`
+
+	// A JWTClaim
+	JWTClaim JWTClaim `yaml:"jwt_claim,omitempty" json:"jwt_claim,omitempty"`
+
+	// String modification to strip off matched value (e.g. "Bearer " for Authorization
+	// tokens).
+	Modification StringModification `yaml:"modification,omitempty" json:"modification,omitempty"`
```

Comment:
Nit: I'm flip-flopping between `Modification` and `Transformation`?

Reply:
`Transformation` sounds cooler to me :)

Reply:
+1 Transformation

Reply:
Ok. So we have the majority :)